recipes/fmilibrary/all/conanfile.py

```diff
@@ -0,0 +1,148 @@
+
+    def requirements(self):
+        self.requires("fmi1/1.0.1")
+        self.requires("fmi2/2.0.4")
+        if self.version >= Version("3.0a2"):
+            self.requires("fmi3/3.0.1")
+        self.requires("expat/2.4.9")
+        self.requires("minizip/1.2.13")
+        self.requires("zlib/[>=1.2.11 <2]")
+        # c99_snprintf -> should be externalised
+
+    def validate(self):
+        if self.settings.os == "Macos" and self.settings.arch == "armv8":
+            raise ConanInvalidConfiguration(
+                f"{self.ref} does not support architecture "
+                f"'{self.settings.arch}' on {self.settings.os}")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+
+        copy(self, "fmiModel*.h", self.dependencies["fmi1"].cpp_info.components["modex"].includedirs[0],
+             path.join(self.build_folder, "fmis", "FMI1"))
+        copy(self, "fmiPlatformTypes.h", self.dependencies["fmi1"].cpp_info.components["cosim"].includedirs[0],
+             path.join(self.build_folder, "fmis", "FMI1"))
+        copy(self, "fmiFunctions.h", self.dependencies["fmi1"].cpp_info.components["cosim"].includedirs[0],
+             path.join(self.build_folder, "fmis", "FMI1"))
+        copy(self, "*.h", self.dependencies["fmi2"].cpp_info.includedirs[0],
+             path.join(self.build_folder, "fmis", "FMI2"))
+
+        tc = CMakeToolchain(self)
+        tc.variables["FMILIB_BUILD_STATIC_LIB"] = not self.options.shared
+        tc.variables["FMILIB_BUILD_SHARED_LIB"] = self.options.shared
+        tc.variables["FMILIB_BUILD_TESTS"] = self.options.with_fmus
+        tc.variables["FMILIB_FMI_STANDARD_HEADERS"] = posixpath.join(self.build_folder, "fmis").replace("\\", "/")
+        tc.variables["FMILIB_GENERATE_DOXYGEN_DOC"] = False
+
+        # The variable is an option only if the following condition is true
+        if not self.options.shared and not self.settings.os in ["Windows", "Macos"]:
+            tc.variables["FMILIB_BUILD_FOR_SHARED_LIBS"] = self.options.get_safe("fPIC", False)
+
+        if is_msvc(self):
+            tc.variables["FMILIB_BUILD_WITH_STATIC_RTLIB"] = is_msvc_static_runtime(self)
+
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0026"] = "OLD"
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0045"] = "OLD"
+        tc.cache_variables["CMAKE_POLICY_DEFAULT_CMP0046"] = "OLD"
+
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+
+        vre = VirtualRunEnv(self)
+        vre.generate(scope="build")
+
+    def build(self):
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+        if not self.conf.get("tools.build:skip_test", default=True):
+            env = Environment()
+            env.define("CTEST_OUTPUT_ON_FAILURE", "ON")
+            with env.vars(self).apply():
+                cmake.test()
```

Comment:
```suggestion
```
Please, try to avoid unit test in recipes. The main idea is packaging a project, not running its unit tests. But in case you really need/want it, use https://docs.conan.io/2/reference/tools/env/envvars.html#envvars with the CMakeToolchain, so you can use a separated script and does not propagate to your environment.